### ISSUE TITLE
`LIBPATCH` limitation on number forces artificial `LIBAPI` updates which mean every consuming charm needs updating

### ISSUE THREAD

**Issue by mthaddon** [canonical/charmcraft#928]
If I have a library that has a `LIBPATCH` number greater than 10 and I try to publish it I get:
```
$ charmcraft publish-lib charms.nginx_ingress_integrator.v0.ingress
Library charms.nginx_ingress_integrator.v0.ingress has a wrong LIBPATCH number, it's too high, Charmhub highest version is 0.10.
```
However, if I increment the `LIBAPI` then every consuming charm will need to specifically know they need to switch from `v0` of my library to `v1`.

This seems very limiting, as when working on a library, and particularly the documentation for a library, it's very common to need multiple updates. Requiring every consuming charm to switch to a newer version when the contents of the library haven't changed substantially seems unusually user-unfriendly.

This is for the nginx-ingress-integrator charm, which is used by lots of charms, so the impact is quite wide.

Could the limitation on the `LIBPATCH` number be removed?

**Comment by mthaddon** [comment]
Sorry, it turns out this was because I was using a non-incremented number (trying to publish with 0.12 when the latest published version was 0.10).

**Comment by facundobatista** [comment]
I created PR #934 to enhance the error message.
